src/controllers/api.controller.js

```diff
@@ -7,4 +9,34 @@ export default {
 		const filePath = `/img/${req.file.filename}`;
 		res.status(200).json({ responseText: "Uploaded", location: filePath });
 	},
+
+	async getCategories(req, res) {
+		try {
+			// Lấy danh sách tất cả các chuyên mục cấp 1 (parent)
+			const parentCategories = await CategoryService.findAllCategories().where("belong_to", null);
+
+			// Thêm các chuyên mục con (subcategories) vào từng chuyên mục cha
+			const categoriesWithSubcategories = await Promise.all(
+				parentCategories.map(async function (category) {
+					const subcategories = await CategoryService.findSubcategories(category.category_id);
+					return {
+						...category,
+						subcategories,
+					};
+				})
+			);
```

Comment:
Already have a service to do this, check article service, the one with the longest name

Reply:
Sorry, the service returns articles, not category, mb